PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -27,15 +27,29 @@ using namespace o2::framework;
 using namespace o2::framework::expressions;
 using namespace o2::aod;
 
+namespace
+{
+static const int nCharmHad = 7;
+static const std::array<int, nCharmHad> PDGArrayParticle = {pdg::Code::kDPlus, 413, pdg::Code::kD0, 431, pdg::Code::kLambdaCPlus, pdg::Code::kXiCPlus, pdg::Code::kJpsi};
+static const std::array<std::string, nCharmHad> labels = {"D^{#plus}", "D*^{#plus}", "D^{0}", "D_{s}^{#plus}", "#Lambda_{c}^{#plus} #rightarrow pK^{#minus}#pi^{#plus}", "#Xi_{c}^{#plus} #rightarrow pK^{#minus}#pi^{#plus}", "J/#psi #rightarrow e^{#plus}e^{#minus}"};
+static const std::array<std::string, nCharmHad> particleNames = {"Dplus", "Dstar", "D0", "Ds", "Lc2pKpi", "Xic2pKpi", "Jpsi2ee"};
+} // namespace
+
 /// Gen Level Validation
 ///
 /// - Number of HF quarks produced per collision
 /// - Number of D±      → π± K∓ π±        per collision
 ///             D*±     → π± K∓ π±,
+///             Ds±     → K± K∓ π±,
 ///             D0(bar) → π± K∓,
 ///             Λc±     → p(bar) K∓ π±
+///             Ξc±     → p(bar) K∓ π±
+///             J/psi   → e∓ e±
```

Comment:
Could be worth to reorder these channels as they enter the arrays defined above, just for clarity